### PR TITLE
Add device conditions to automation editor.

### PR DESCRIPTION
**Pull request by emontnemery**
This depends on home-assistant/home-assistant#26313

Screenshot:
image

### CODE REVIEW

File: src/components/device/ha-device-automation-picker.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    );
+
+    if (idx === -1) {
+      return UNKNOWN_AUTOMATION_KEY;
+    }
+
+    return `${this._automations[idx].device_id}_${idx}`;
+  }
+
+  protected localizeDeviceAutomation(value) {
```
Comment (emontnemery) [inline src/components/device/ha-device-automation-picker.ts]: I want to do `protected abstract localizeDeviceAutomation(value);`, but then the call to this.localizeDeviceAutomation() from render() does not work.

File: src/components/device/ha-device-automation-picker.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      this._setValue(
+        this._automations.length ? this._automations[0] : this._noAutomation
+      );
+    }
+    this._renderEmpty = true;
+    await this.updateComplete;
+    this._renderEmpty = false;
+  }
+
+  protected get _noAutomation(): T {
```
Comment (emontnemery) [inline src/components/device/ha-device-automation-picker.ts]: Same issue with abstract method as above.

File: src/components/device/ha-device-automation-picker.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    // The value has changed, force the listbox to update
+    if (changedProps.has("value") || changedProps.has("_renderEmpty")) {
+      const listbox = this.shadowRoot!.querySelector("paper-listbox")!;
+      if (listbox) {
+        listbox._selectSelected(this._key);
+      }
+    }
+  }
+
+  protected async _fetchDeviceAutomations() {
```
Comment (emontnemery) [inline src/components/device/ha-device-automation-picker.ts]: Same issue with abstract method as above.

File: src/data/device_automation.ts [hunk 4]
```diff
@@ -1,30 +1,45 @@
...
+export interface DeviceConditionList {
+  conditions: DeviceCondition[];
+}
+
 export interface DeviceTriggerList {
   triggers: DeviceTrigger[];
 }
 
+export const fetchDeviceConditions = (hass: HomeAssistant, deviceId: string) =>
+  hass.callWS<DeviceConditionList>({
```
Comment (balloob) [inline src/data/device_automation.ts]: Shouldn't this be 

```suggestion
  hass.callWS<DeviceCondition[]>({
```

File: src/data/device_automation.ts [hunk 5]
```diff
@@ -1,30 +1,45 @@
...
 
+export interface DeviceCondition extends DeviceAutomation {
+  condition: string;
+}
+
+export interface DeviceTrigger extends DeviceAutomation {
+  platform: string;
+}
+
+export interface DeviceConditionList {
```
Comment (balloob) [inline src/data/device_automation.ts]: can be removed.

File: src/data/device_automation.ts [hunk 6]
```diff
@@ -1,30 +1,45 @@
...
+  conditions: DeviceCondition[];
+}
+
 export interface DeviceTriggerList {
```
Comment (balloob) [inline src/data/device_automation.ts]: can be removed

File: src/data/device_automation.ts [hunk 7]
```diff
@@ -1,30 +1,45 @@
...
-    .callWS<DeviceTriggerList>({
-      type: "device_automation/trigger/list",
-      device_id: deviceId,
-    })
-    .then((response) => response.triggers);
-
-export const deviceAutomationTriggersEqual = (
-  a: DeviceTrigger,
-  b: DeviceTrigger
+  hass.callWS<DeviceTriggerList>({
```
Comment (balloob) [inline src/data/device_automation.ts]:
```suggestion
  hass.callWS<DeviceTrigger[]>({
```

Comment (emontnemery) [reply src/data/device_automation.ts]: Fixed.

File: src/data/device_automation.ts (hunk 6 above)
Comment (emontnemery) [reply src/data/device_automation.ts]: Fixed

File: src/data/device_automation.ts (hunk 5 above)
Comment (emontnemery) [reply src/data/device_automation.ts]: Fixed

File: src/data/device_automation.ts (hunk 4 above)
Comment (emontnemery) [reply src/data/device_automation.ts]: Yeah, fixed.

**Pull request merged**